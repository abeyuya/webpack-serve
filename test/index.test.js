const request = require('supertest');

const serve = require('../lib/index');

describe('serve', () => {
  test('serve', () => {
    const argv = { logLevel: 'silent' };
    const opts = {};
    return serve(argv, opts).then(({ app, options }) => {
      expect(app).toMatchSnapshot();
      expect(options).toMatchSnapshot();

      return new Promise((resolve) => {
        setTimeout(() => app.stop(resolve), 500);
      });
    });
  });

  test('serve basic config', () => {
    const argv = { logLevel: 'silent' };
    const opts = { config: require('./fixtures/basic/webpack.config') };
    return serve(argv, opts).then(({ app, on, options }) => {
      const { server } = app;
      expect(app).toMatchSnapshot();
      expect(options).toMatchSnapshot();

      return new Promise((resolve) => {
        on('build-finished', ({ stats }) => {
          expect(stats).toBeDefined();
          request(server)
            .get('/output.js')
            .expect(200)
            .then((response) => {
              expect(response.text).toMatchSnapshot();
              app.stop(resolve);
            });
        });
      });
    });
  });

  test('serve config + add', () => {
    const argv = { logLevel: 'silent' };
    const opts = {
      add(app, middleware) {
        middleware.webpack();

        middleware.content({
          index: 'index.htm',
        });
      },
      config: require('./fixtures/htm/webpack.config'),
    };

    return serve(argv, opts).then(({ app }) => {
      const req = request(app.server);
      return Promise.all([
        req.get('/index.htm').expect(200),
        req.get('/output.js').expect(200),
      ]).then(() => new Promise((resolve) => app.stop(resolve)));
    });
  });

  test('serve multi config', () => {
    const argv = { logLevel: 'silent' };
    const opts = { config: require('./fixtures/multi/webpack.config') };
    return serve(argv, opts).then(({ app }) => {
      const req = request(app.server);
      return Promise.all([
        req.get('/static/client.js').expect(200),
        req.get('/server/server.js').expect(200),
      ]).then(() => new Promise((resolve) => app.stop(resolve)));
    });
  });

  test('serve multi-named config', () => {
    const argv = { logLevel: 'silent' };
    const opts = { config: require('./fixtures/multi-named/webpack.config') };
    return serve(argv, opts).then(({ app }) => {
      const req = request(app.server);
      return Promise.all([
        req.get('/bundle1.js').expect(200),
        req.get('/bundle2.js').expect(200),
      ]).then(() => new Promise((resolve) => app.stop(resolve)));
    });
  });

  test('serve webpack 4 defaults config', () => {
    const argv = { logLevel: 'silent' };
    const opts = {
      config: require('./fixtures/webpack-4-defaults/webpack.config'),
    };
    return serve(argv, opts).then(({ app }) => {
      const req = request(app.server);
      return Promise.all([
        req.get('/index.html').expect(200),
        req.get('/main.js').expect(200),
      ]).then(() => new Promise((resolve) => app.stop(resolve)));
    });
  });
});
