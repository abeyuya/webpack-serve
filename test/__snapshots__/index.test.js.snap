// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serve basic config 1`] = `
Object {
  "env": "test",
  "proxy": false,
  "subdomainOffset": 2,
}
`;

exports[`serve basic config 2`] = `
Object {
  "bus": Nanobus {
    "_listeners": Object {},
    "_name": "nanobus",
    "_starListeners": Array [],
  },
  "clipboard": true,
  "compiler": Any<Object>,
  "content": Array [
    "<PROJECT_ROOT>/test/fixtures/basic",
    "<PROJECT_ROOT>",
  ],
  "devMiddleware": Object {
    "logLevel": "silent",
    "publicPath": "/",
  },
  "host": "localhost",
  "hotClient": Object {
    "host": "localhost",
    "logLevel": "silent",
  },
  "http2": false,
  "https": null,
  "logLevel": "silent",
  "logTime": false,
  "open": false,
  "port": 8080,
  "protocol": "http",
}
`;

exports[`serve basic config 3`] = `
"!function(e){var n={};function t(o){if(n[o])return n[o].exports;var r=n[o]={i:o,l:!1,exports:{}};return e[o].call(r.exports,r,r.exports,t),r.l=!0,r.exports}t.m=e,t.c=n,t.d=function(e,n,o){t.o(e,n)||Object.defineProperty(e,n,{enumerable:!0,get:o})},t.r=function(e){\\"undefined\\"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:\\"Module\\"}),Object.defineProperty(e,\\"__esModule\\",{value:!0})},t.t=function(e,n){if(1&n&&(e=t(e)),8&n)return e;if(4&n&&\\"object\\"==typeof e&&e&&e.__esModule)return e;var o=Object.create(null);if(t.r(o),Object.defineProperty(o,\\"default\\",{enumerable:!0,value:e}),2&n&&\\"string\\"!=typeof e)for(var r in e)t.d(o,r,function(n){return e[n]}.bind(null,r));return o},t.n=function(e){var n=e&&e.__esModule?function(){return e.default}:function(){return e};return t.d(n,\\"a\\",n),n},t.o=function(e,n){return Object.prototype.hasOwnProperty.call(e,n)},t.p=\\"\\",t(t.s=0)}({\\"../../../node_modules/webpack/buildin/module.js\\":function(e,n){e.exports=function(e){return e.webpackPolyfill||(e.deprecate=function(){},e.paths=[],e.children||(e.children=[]),Object.defineProperty(e,\\"loaded\\",{enumerable:!0,get:function(){return e.l}}),Object.defineProperty(e,\\"id\\",{enumerable:!0,get:function(){return e.i}}),e.webpackPolyfill=1),e}},\\"./app.js\\":function(e,n,t){(function(e){t(\\"./component.js\\"),\\"object\\"==typeof e.hot&&e.hot.accept(e=>{e&&console.error(\\"Cannot apply HMR update.\\",e)})}).call(this,t(\\"../../../node_modules/webpack/buildin/module.js\\")(e))},\\"./component.js\\":function(e,n){document.querySelector(\\"#target\\").innerHTML=\\"component\\"},0:function(e,n,t){e.exports=t(\\"./app.js\\")}});
//# sourceMappingURL=output.js.map"
`;

exports[`serve serve 1`] = `
Object {
  "env": "test",
  "proxy": false,
  "subdomainOffset": 2,
}
`;

exports[`serve serve 2`] = `
Object {
  "bus": Nanobus {
    "_listeners": Object {},
    "_name": "nanobus",
    "_starListeners": Array [],
  },
  "clipboard": true,
  "compiler": Any<Object>,
  "content": Array [
    "<PROJECT_ROOT>",
  ],
  "devMiddleware": Object {
    "logLevel": "silent",
    "publicPath": "/",
  },
  "host": "localhost",
  "hotClient": Object {
    "host": "localhost",
    "logLevel": "silent",
  },
  "http2": false,
  "https": null,
  "logLevel": "silent",
  "logTime": false,
  "open": false,
  "port": 8080,
  "protocol": "http",
}
`;
